GitCommands/RevisionReader.cs

```diff
@@ -240,19 +240,30 @@ private IReadOnlyCollection<GitRevision> GetRevisionsFromArguments(GitArgumentBu
             using IProcess process = _module.GitCommandRunner.RunDetached(cancellationToken, arguments, redirectOutput: true, outputEncoding: GitModule.LosslessEncoding);
             cancellationToken.ThrowIfCancellationRequested();
 
+            // Initial buffer to give very quick feedback to user
+            List<GitRevision> revisions = new(100);
             foreach (ReadOnlyMemory<byte> chunk in process.StandardOutput.BaseStream.SplitLogOutput())
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (TryParseRevision(chunk.Span, out GitRevision? revision))
+                if (TryParseRevision(chunk, out GitRevision? revision))
                 {
 #if TRACE_REVISIONREADER
                     revisionCount++;
 #endif
-                    subject.OnNext(revision);
+                    revisions.Add(revision);
+                    if (revisions.Count == revisions.Capacity)
+                    {
+                        subject.OnNext(revisions);
+
+                        // ... then use big buffer to load all the remaining revisions with better performance
+                        revisions = new(100_000);
```

Comment:
I suggest slightly smaller chunks (and to clear, there may be more than 100K revisions).
```suggestion
                        if (revisions.Capacity < 10_000)
                        {
                            revisions = new(10_000);
                        }
                        else
                        {
                            revisions.Clear();
                        }
```

Reply:
> I suggest slightly smaller chunks

👍 but independently from the `Capacity`
The grid stays empty until all revisions of the Linux repo have been loaded.
Filters can heavily slow down the loading. Then I want to see every single revision being added in the visible range.

Please be explicit:
`const int minimumInitialCapacity = 100_000;`
How much memory does it cost for a small repo in comparison to `10_000`?

Reply:
> > I suggest slightly smaller chunks
> 
> 👍 but independently from the `Capacity`
> **The grid stays empty until all revisions of the Linux repo have been loaded.** Filters can heavily slow down the loading. Then I want to see every single revision being added in the visible range.

TL;DR: At least WorkDir and Index are inserted at the wrong position.

I test with loading all commits. The most recent commit of this repo is from 2023/02/25; the current checkout is from 2023/01/13.
The current **master build** shows a blank grid for more than 30 seconds, too, and traces:

on open with Linux repo

![image](https://github.com/gitextensions/gitextensions/assets/36601201/70eddd97-7a5a-4974-ba7a-747bd57c023e)

on switch to Linux repo (shows the current checkout quickly)

![image](https://github.com/gitextensions/gitextensions/assets/36601201/8a251bb7-11bb-4231-a032-eae4c7c529ec)

and shows WorkDir and Index at the correct position:

![image](https://github.com/gitextensions/gitextensions/assets/36601201/582aca26-5b44-4bc6-9f7c-531a2caa1694)

**This PR shows** a blank grid for more than 30 seconds and traces:

on open with Linux repo

![image](https://github.com/gitextensions/gitextensions/assets/36601201/70d93bc9-b50d-4fd3-b36b-2f14b7578606)

on switch to Linux repo (shows the current checkout quickly)

![image](https://github.com/gitextensions/gitextensions/assets/36601201/5003292f-3ee9-44f0-888f-0dd7d467a8ba)

but shows WorkDir and Index dozens of pages above the checked-out commit

![image](https://github.com/gitextensions/gitextensions/assets/36601201/a1ecd4d2-b37e-4903-9ef9-354194e91cca)